src/main.cpp

```diff
@@ -2242,6 +2265,110 @@ bool Session::Initialize(const std::uint8_t config, uint32_t max_user_bandwidth)
   return InitializePeriod();
 }
 
+
+bool Session::PreInitializeDRM(std::string& challengeB64, std::string& sessionId)
+{
+    std::string psshData;
+    std::string kidData;
+    // Parse the PSSH/KID data
+    std::string::size_type posSplitter(drmPreInitData_.find("|"));
+    if (posSplitter != std::string::npos)
+    {
+      psshData = drmPreInitData_.substr(0, posSplitter);
+      kidData = drmPreInitData_.substr(posSplitter + 1);
+    }
+
+    if (psshData.empty() || kidData.empty())
+    {
+      kodi::Log(ADDON_LOG_ERROR, "%s - Invalid DRM pre-init data, must be as: {PSSH as base64}|{KID as base64}", __FUNCTION__);
+      return false;
+    }
+
+    cdm_sessions_.resize(2); //(adaptiveTree_->current_period_->psshSets_.size());  TODO: One session for a single period/PSSH, why is needed size of 2?
```

Comment:
@matthuisman i have a question on this line,
as you can also see in the `Session::InitializeDRM` method the cdm_sessions_ variable is resized, but, why the first value of the array is not used?

Reply:
Periods always have a `psshSets[0]` by default (I think it's just used for AES-128) - lots of logic is relying on this i.e https://github.com/xbmc/inputstream.adaptive/blob/a675541a5c10e0ff4e0ec2fa7b4c9d8306bcffb8/src/main.cpp#L2291
https://github.com/xbmc/inputstream.adaptive/blob/a675541a5c10e0ff4e0ec2fa7b4c9d8306bcffb8/src/main.cpp#L2436
https://github.com/xbmc/inputstream.adaptive/blob/a675541a5c10e0ff4e0ec2fa7b4c9d8306bcffb8/src/common/AdaptiveTree.cpp#L286-L297